app/code/Magento/InventoryFrontendUi/Plugin/Block/Stockqty/AbstractStockqtyPlugin.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: alessandro
+ * Date: 09/06/18
+ * Time: 15.35
+ */
+namespace Magento\InventoryFrontendUi\Plugin\Block\Stockqty;
+
+use Magento\InventoryConfigurationApi\Api\GetStockItemConfigurationInterface;
+use Magento\InventorySalesApi\Api\GetProductSalableQtyInterface;
+use Magento\InventorySalesApi\Model\StockByWebsiteIdResolverInterface;
+
+class AbstractStockqtyPlugin
+{
+    /**
+     * @var GetStockItemConfigurationInterface
+     */
+    private $getStockItemConfiguration;
+
+    /**
+     * @var StockByWebsiteIdResolverInterface
+     */
+    private $stockByWebsiteId;
+
+    /**
+     * @var GetProductSalableQtyInterface
+     */
+    private $getProductSalableQty;
+
+    public function __construct(
+        StockByWebsiteIdResolverInterface $stockByWebsiteId,
+        GetStockItemConfigurationInterface $getStockItemConfig,
+        GetProductSalableQtyInterface $getProductSalableQty
+    ) {
+        $this->getStockItemConfiguration = $getStockItemConfig;
+        $this->stockByWebsiteId = $stockByWebsiteId;
+        $this->getProductSalableQty = $getProductSalableQty;
+    }
+
+    /**
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function aroundIsMsgVisible(
+        \Magento\CatalogInventory\Block\Stockqty\AbstractStockqty $subject,
+        callable $proceed
+    ): bool {
+        $sku = $subject->getProduct()->getSku();
+        $websiteId = $subject->getProduct()->getStore()->getWebsiteId();
+        $stockId = $this->stockByWebsiteId->execute($websiteId)->getStockId();
+        $stockItemConfig = $this->getStockItemConfiguration->execute($sku, $stockId);
+        if (null === $stockItemConfig) {
+            return false;
+        }
+        return $this->getProductSalableQty->execute($sku, $stockId) > 0
```

Comment:
let's use clean check for Backorder
`getBackorders` instead of `$this->getProductSalableQty->execute($sku, $stockId) > 0`
because even if we have Salable Qty = 5 , and getStockThresholdQty = 10
that's not correct to say that only 5 left, because having backorder enabled we can sell infinite number

Reply:
I'm not sure I understood your proposal, I changed the code like this:
```
return $stockItemConfig->getBackorders() === StockItemConfigurationInterface::BACKORDERS_NO
            && $this->getProductSalableQty->execute($sku, $stockId) > 0
            && $this->getStockQtyLeft($sku, $stockId) <= $stockItemConfig->getStockThresholdQty();
```
If backorders are enabled, the first condition will fail and the message won't be dispayed.

If backorders are disabled, the first condition will return `true` and the other two conditions check that the product is salable and that the salable qty is lett or equal than the threshold.

Is that correct?